File: Heat/Validation/ValidationRules.vb
Public Class ValidationRules
    Inherits List(Of IValidator)

    Public ReadOnly Property IsValid() As Boolean
        Get
            Return Not IsNothing(Me) AndAlso Me.All(Function(x) x.IsValid)
        End Get
    End Property

    Public ReadOnly Property Errors As List(Of IValidator)
        Get
            Return Me.Where(Function(x) Not x.IsValid).ToList
        End Get
    End Property

End Class

File: Heat.ConvertedToC#/Validation/ValidationRules.cs
using System.Collections.Generic;
using System.Linq;
namespace Heat
{
    public class ValidationRules : List<IValidator>
	{

		public bool IsValid {
			get { return (this != null) && this.All(x => x.IsValid); }
		}

		public List<IValidator> Errors {
			get { return this.Where(x => !x.IsValid).ToList(); }
		}

	}
}
